projects/plugins/jetpack/class.jetpack.php

```diff
@@ -4149,21 +4149,13 @@ public function deactivate_dialog( $hook ) {
 					true
 				);
 
-				wp_register_script(
-					'jp-tracks-functions',
-					plugins_url( '_inc/lib/tracks/tracks-callables.js', JETPACK__PLUGIN_FILE ),
-					array( 'jp-tracks' ),
-					JETPACK__VERSION,
-					false
-				);
-
 				wp_enqueue_script(
 					'jetpack-deactivate-dialog-js',
 					Assets::get_file_url_for_environment(
 						'_inc/build/jetpack-deactivate-dialog.min.js',
 						'_inc/jetpack-deactivate-dialog.js'
 					),
-					array( 'jquery', 'jp-tracks-functions' ),
```

Comment:
I'm slightly confused. Shouldn't this dependable still be required? The issue is that it should be enqueued from the package but this jetpack-deactivate-dialog-js still is dependent on it.

Same with jetpack-search-widget-admin below.

Or is this me forgetting how all these wp_register_scripts work.

Reply:
These questions are also relevant to my interests :)
I'm not fully understanding how this all works. Originally, I thought it was as simple as adding this tracks-callable.js file to the Tracking package... but no, there's more to it than that :)

Reply:
In my opinion, yes, this dependency should remain here. I guess the original intention behind `wp_register_script` and actual `wp_enqueue_script` was that the first one would only add the script and its path to a list of available files to be loaded, like `jquery`, and then once a script gets enqueued that actually needs that script and lists it as its dependency, it gets added to the output.

We can do something similar here by introducing a new `register_tracks_scripts` function in the Tracks class, and then calling it from packages that need that file. Sort of like this:

```
Tracks::register_scripts();
wp_enqueue_script( ...., array( 'jquery', 'jp-tracks-functions' )
```

That `jp-tracks-functions` name we can also make more verbose to keep it in line with the whole namespacing thing. For example, make it `jetpack-tracks-functions`. Sort of like `plugin-package-file`.